### PR TITLE
[SYCL] Add tests for `atomic_memory_order/scope` context queries.

### PR DESCRIPTION
**Pull request by maarquitos14**
Add tests for `info::context::atomic_memory_scope_capabilities` and `info::context::atomic_memory_order_capabilities` queries.

### CODE REVIEW

File: sycl/source/detail/context_impl.cpp [hunk 1]
```diff
@@ -193,17 +193,26 @@ template <>
...
+    std::vector<sycl::memory_scope> NewCapabilityList;
+    std::vector<sycl::memory_scope> DeviceCapabilities =
+        Device.get_info<info::device::atomic_memory_scope_capabilities>();
+    std::set_intersection(
+        CapabilityList.begin(), CapabilityList.end(),
+        DeviceCapabilities.begin(), DeviceCapabilities.end(),
+        std::inserter(NewCapabilityList, NewCapabilityList.begin()));
+    CapabilityList = NewCapabilityList;
+  }
+  CapabilityList.shrink_to_fit();
```
[the remark this answers is not in the document]
Comment (maarquitos14) [reply sycl/source/detail/context_impl.cpp]: This was addressed in the latest update.

**Comment by maarquitos14** [comment]
Friendly ping @aelovikov-intel @intel/llvm-reviewers-runtime @intel/sycl-language-enabling-triage

**Comment by KornevNikita** [comment]
Hi @maarquitos14, I see only tests changes here. Is that how it's intended?

**Comment by maarquitos14** [comment]
> Hi @maarquitos14, I see only tests changes here. Is that how it's intended?

I just realized the changes in context_impl.cpp were already merged in #8586. I guess I accidentally included the changes there too. Thus, we only have tests in this PR now. I'll update the PR title accordingly.

**Review by aelovikov-intel — APPROVED** [review]

**Comment by maarquitos14** [comment]
Friendly ping @intel/sycl-language-enabling-triage

**Pull request merged**